app/javascript/src/components/Dashboard/Contacts/index.jsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect } from "react";
+
+import EmptyNotesListImage from "images/EmptyNotesList";
+import { PageLoader } from "neetoui/v2";
+import { Container } from "neetoui/v2/layouts";
+
+import EmptyState from "components/Common/EmptyState";
+
+import { SAMPLE_CONTACTS } from "./constants";
+import ContactTable from "./ContactTable";
+import SideMenu from "./SideMenu";
+import TopHeader from "./TopHeader";
+
+const Notes = () => {
+  const [loading, setLoading] = useState(true);
+  const [showSideMenu, setShowSideMenu] = useState(true);
+  const [contacts, setContacts] = useState([]);
+
+  useEffect(() => {
+    fetchContacts();
+  }, []);
+
+  const fetchContacts = () => {
+    try {
+      setLoading(true);
+      setContacts(SAMPLE_CONTACTS);
+    } catch (error) {
+      logger.error(error);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  if (loading) {
+    return <PageLoader />;
+  }
+
+  return (
+    <>
+      <SideMenu isOpen={showSideMenu} />
+      {contacts.length ? (
+        <Container>
+          <TopHeader toggleSideMenu={() => setShowSideMenu(prev => !prev)} />
```

Comment:
What's your reasoning for using prevState from within the setter over here rather than using `!showSideMenu`? What are the advantages of using this pattern over here?

Reply:
I had thought this was the better practice, sorry.

Reply:
@shamilsdq Here we should use `prevState`. The question was on why you thought it was a better practice to use it in the first place. If you don't know the reason, then no issues I can help you out. But make sure to let me know.